app/models/vm_or_template.rb

```diff
@@ -1833,6 +1837,14 @@ def parent_resource
     parent
   end
 
+  def available_child_options
+    Rbac.filtered(VmOrTemplate.active.by_ems.where.not(:id => id).select { |vm| vm.parent.nil? })
```

Comment:
`select { |vm| vm.parent.nil? })` I think this is N+1 - maybe you can create query with `Relationship` as was removed here (https://github.com/ManageIQ/manageiq-ui-classic/pull/2271/files#diff-5910cce3bec4cd624ba288de86215b4cL1406)

Reply:
and you probably wanted to add parameter to `.by_ems`